### ISSUE TITLE
Podman: "Layer not known" error 

### ISSUE THREAD

**Issue by jomeier** [openshift/okd#297]
Hi,

on vSphere in our company and on my homelab I've had a problem recently where podman reports "Layer not known" after showing the images with "sudo podman images".

This is not only a "visual" bug but also crio can't access the images and therefore can't start containers. No containers are running anymore: crictl ps -a shows an empty list.

```
[core@chmu3w0 ~]$ sudo podman images
ERRO[0000] error checking if image is a parent "70a981a7a6de09f373771eb639ad600dd30cb9bb1de96cee2b92f6940807d44b": layer not known
ERRO[0000] error checking if image is a parent "659694877391ed9f53dcefdeed267c5f8dfd24154ef6abe3c1f1be9c54a34088": layer not known
ERRO[0000] error checking if image is a parent "4665c12a232572c6f18d7594fdd9c68b8b8c05a95f4c81ab3f62c74469f4ca78": layer not known
ERRO[0000] error checking if image is a parent "e6de8ebc835401112ba0657e672ad187acaf77754456f9fec7ba04401f80f314": layer not known
REPOSITORY                                                                                  TAG                            IMAGE ID       CREATED         SIZE

docker.io/minio/minio                                                                       latest                         4665c12a2325   3 weeks ago     unable to determine size: error locating layer with ID "4bb81fe5da12b12e168842c264247bfb520c43a19374d720f3c3f8c2ec574b2b": layer not known
quay.io/openshift/okd-content                                                               <none>                         256bc3e8da88   3 weeks ago     unable to determine size: error locating layer with ID "2aadc2ab4ea6a3598b45efba862a5d521e9554a2e73893c7f76fa34f9b46c457": layer not known
quay.io/openshift/okd-content                                                               <none>                         b278aa4f2719   4 weeks ago     unable to determine size: error locating layer with ID "0f003470a3208f897ceb401ab7df52c0e1a0eb8bf8e4b2ac6f6b540da450cea8": layer not known
quay.io/openshift/okd-content                                                               <none>                         ebb673575b8b   4 weeks ago
```

This seems to be an old problem with podman. Any ideas how we can fix this ? This seems to happen randomly if a node restarts. But we have also seen that without having nodes restarted.

Temprorary solution:

Delete the directory: /var/lib/containers/storage/overlay-images/  

After restarting the node, the images get pulled again and the node gets ready afterwards.

Greetings,

Josef

**Comment by vrutkovs** [comment]
Looks very similar to https://bugzilla.redhat.com/show_bug.cgi?id=1857224 - so next cri-o update might fix that. It appears that next FCOS stable update (scheduled ~Aug 25) would resolve blocking [NFS bug](https://bugzilla.redhat.com/show_bug.cgi?id=1854379) and new CRI-O update would be pulled in

**Comment by jomeier** [comment]
Ok. That is worth a try!

**Comment by vrutkovs** [comment]
FCOS bump didn't resolve the NFS blocking bug, not sure why :(

Its certainly fixed in kernel 5.8, so we're waiting for another FCOS stable update

**Comment by vrutkovs** [comment]
Fixed in latest OKD 4.5 nighlies (with CRI-O 1.18.4) and OKD 4.6 (using CRI-O 1.19)
